cardinal/system/t_meat_puppet_system_test.go

```diff
@@ -0,0 +1,290 @@
+
+// TestInsultMeat tests the insultMeat function
+func TestInsultMeat(t *testing.T) {
+	tests := []struct {
+		name        string
+		cErr        uint8
+		badCmd      string
+		expectedMsg string
+	}{
+		{
+			name:        "Too many tokens",
+			cErr:        constants.ErrParserRoutineTKCX.Code,
+			badCmd:      "",
+			expectedMsg: "WTF, slow down cowboy, you're gonna hurt yourself",
+		},
+		{
+			name:        "Gibberish command",
+			cErr:        constants.ErrParserRoutineNOP.Code,
+			badCmd:      "",
+			expectedMsg: "Nope, gibberish. Stop breathing with your mouth.",
+		},
+		{
+			name:        "No direction",
+			cErr:        constants.ErrParserRoutineND.Code,
+			badCmd:      "north",
+			expectedMsg: "Go where pilgrim?",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			msg := insultMeatTest(tt.cErr, tt.badCmd)
+			assert.Equal(t, tt.expectedMsg, msg)
+		})
+	}
+}
+
+// Process the Commands tokens, this is the function dedicated to it.
+func ProcessCommandsTokensLogicTest(Tokens []string, PlayerID uint32, roomID uint32) (string, bool, uint8) {
+	pID := PlayerID
+	rID := roomID
+	tokens := Tokens
+	var er uint8
+	var move bool
+	var output string
+	//var nxt uint32 ---> Not used YET
+
+	// Start a new token system
+	ts = NewTokeniserSystem()
+
+	if uint8(len(tokens)) > constants.MAX_TOK {
+		er = constants.ErrParserRoutineTKCX.Code
+		output = constants.ErrParserRoutineTKCX.Message
+	} else {
+		var tok1 string
+		tok1 = tokens[0]
+		tokD := ts.GetDirectionType(tok1)
+
+		if tokD != enums.DirectionTypeNone {
+			move = true
+			// HERE GOES GET NEXT ROOM - DIRECTION SYSTEM
+		} else if ts.GetActionType(tok1) != enums.ActionTypeNone {
+			if uint8(len(tokens)) >= constants.MIN_TOK {
+
+				if ts.GetActionType(tok1) == enums.ActionTypeGo {
+					// GO: form
+					move = true
+					output = "GOING TO NEXT ROOM - DIRECTION SYSTEM - TO BE IMPLEMENTED"
+					// HERE GOES GET NEXT ROOM - DIRECTION SYSTEM
+				} else {
+					// VERB: form
+					output, er = handleVerbTest(tokens, rID, pID)
+					move = false
+				}
+
+			} else {
+				er = handleAliasTest(tokens, pID)
+				move = false
+				output = "VERB GOES TO HANDLE ALIAS. TO BE IMPLEMENTED"
+			}
+		} else {
+			er = constants.ErrParserRoutineNOP.Code
+			output = constants.ErrParserRoutineNOP.Message
+		}
+	}
+
+	return output, move, er
+}
+
+func handleAliasTest(tokens []string, playerID uint32) (err uint8) {
+	vrb := ts.GetActionType(tokens[0])
+	//var e uint8
+	if vrb == enums.ActionTypeInventory {
+		fmt.Printf("---->HANDLE ALIAS: NOW SHOULD BE GOING TO INVENTORY FROM INVENTORY SYSTEM")
+		err = 135 // This is just for showing errors
+	} else if vrb == enums.ActionTypeLook {
+		fmt.Printf("--->HANDLE ALIAS: NOW SHOULD BE GOING TO STUFF FROM LOOK SYSTEM")
+		err = 130 // This is just for showing errors
+	}
+	return err
+}
+
+// Handle if the token is a verb
+func handleVerbTest(tokens []string, roomID uint32, playerID uint32) (output string, err uint8) {
```

Comment:
this is not a test its a function to return some stuff to a test so it only tests tests.

Reply:
Leaving this file until get response from Argus guys